Mage/src/main/java/mage/abilities/effects/common/ruleModifying/CombatDamageByToughnessSourceEffect.java

```diff
@@ -0,0 +1,45 @@
+package mage.abilities.effects.common.ruleModifying;
+
+import mage.abilities.Ability;
+import mage.abilities.effects.ContinuousEffectImpl;
+import mage.constants.Duration;
+import mage.constants.Layer;
+import mage.constants.Outcome;
+import mage.constants.SubLayer;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.permanent.PermanentIdPredicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+
+public class CombatDamageByToughnessSourceEffect extends ContinuousEffectImpl {
+    
+    public CombatDamageByToughnessSourceEffect(Duration duration) {
+        super(duration, Layer.RulesEffects, SubLayer.NA, Outcome.Neutral);
+        this.staticText = "{this} assigns combat damage equal to its toughness rather than its power";
+    }
+
+    private CombatDamageByToughnessSourceEffect(final CombatDamageByToughnessSourceEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public CombatDamageByToughnessSourceEffect copy() {
+        return new CombatDamageByToughnessSourceEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent permanent = game.getPermanent(source.getSourceId());
+        if (permanent == null) {
+            return false;
```

Comment:
can `discard()` before returning here as if the permanent is no longer there then the effect won't be valid anytime in the future

Reply:
well, guess I was wrong about this one, oops